lib/puppet/provider/service/systemd.rb

```diff
@@ -145,5 +157,12 @@ def stopcmd
   def statuscmd
     [command(:systemctl), "is-active", @resource[:name]]
   end
+
+  def prepare_error_message(name, action)
+    error_return = "Systemd #{action} for #{name} failed:"
+    journalctl_output = execute("journalctl -n 50 --since '5 minutes ago' -u #{name} --no-pager")
```

Comment:
I'm assuming it would raise a normal Puppet execution error? I could add a spec in for that